File: ruby/reverse_string.rb
def reverse_string(str)
  reversedStr = ''
  i = 0
  while i < str.size
    reversedStr = str[i] + reversedStr

    i += 1
  end
  reversedStr
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {

  // create a new variable to hold a reversed string
  let reversedStr = ''
  
  // iterate through the given string
  for (let i = 0; i < str.length; i ++) {

    // with every iteration add an element from the given string to the BEGINNING of the reversed string
    reversedStr = str[i] + reversedStr
  }

  // return reversed array
  return reversedStr;
}




if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution
